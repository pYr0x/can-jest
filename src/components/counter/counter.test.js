import Counter from "./counter";
import {stache} from "can";

test("counter", () => {
  const counter = new Counter();
  expect(counter.viewModel.count).toBe(0);

  counter.element.querySelector('button').click();
  expect(counter.viewModel.count).toBe(1);
});

test("counter in DOM", () =>{
  const view = stache('<my-counter></my-counter>');

  document.body.innerHTML = `<div id="test"></div>`;
  document.querySelector("#test").appendChild(view());

  const counter = document.querySelector("my-counter span");

  expect(counter.innerHTML).toBe("0");

  document.querySelector('button').click();
  expect(counter.innerHTML).toBe("1");

});

it("test with jest-dom", () => {
  document.body.innerHTML = `<div id="hw"></div>`;
  expect(document.querySelector("#hw")).toBeInTheDocument();
});

test("snapshot", () => {
  const counter = new Counter();
  expect(counter.element).toMatchSnapshot("el");
  expect(counter.viewModel.serialize()).toMatchSnapshot("vm");
});