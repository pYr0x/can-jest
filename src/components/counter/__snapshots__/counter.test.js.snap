// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: el 1`] = `
<my-counter>
  
        Count: 
  <span>
    0
  </span>
  
        
  <button
    on:click="this.increment()"
  >
    +1
  </button>
  
    
</my-counter>
`;

exports[`snapshot: vm 1`] = `
Object {
  "count": 0,
}
`;
